tests/spec/string/spec-interpolate.js

```diff
@@ -68,6 +68,41 @@ define(['mout/string/interpolate'], function (interpolate) {
             expect( interpolate('{{-#$&@_}}', replacements) ).toEqual('foo bar');
         });
 
+
+        it('should support deeply nested templates', function() {
+            var replacements = {
+                'level1': '{{level2}}',
+                'level2': 'This is {{level3}}',
+                'level3': 'somewhat deep'
+            };
+            expect( interpolate('{{level1}} nesting', replacements ) ).toBe( 'This is somewhat deep nesting' );
+            expect( interpolate('{{0}} nesting', ['{{1}}', 'This is {{2}}', 'somewhat deep']) ).toBe( 'This is somewhat deep nesting' );
+        });
+
+
+        it('should support multiple deeply nested templates', function() {
+            var replacements = {
+                'level1' : '{{level2a}}. {{level2b}}',
+                'level2a': 'This sentence is made of {{level3a}}',
+                'level2b': 'And, this one has {{level3b}}',
+                'level3a': 'three {{level4}}',
+                'level3b': 'only two',
+                'level4' : 'nested templates'
+            };
+            expect( interpolate('{{level1}} nested templates.', replacements ) ).toBe( 'This sentence is made of three nested templates. And, this one has only two nested templates.' );
```

Comment:
I think you can just use `include` to match part of the replacements. 

The equal is not so good here. It transform it into a biiiig string.